tools/write_to_sqlite.py

```diff
@@ -0,0 +1,146 @@
+import alara_output_processing as aop
+import numpy as np
+import openmc
+import argparse
+import yaml
+import pandas as pd
+import sqlite3
+from sympy import symbols, Eq, solve
+
+def open_flux_file(flux_file):
+    flux_lines = open(flux_file, 'r').readlines()
+    return flux_lines
+
+def calc_time_params(active_burn_time, duty_cycle_list, num_pulses):
+    t_irr_arr = np.ndarray((len(num_pulses), len(duty_cycle_list)), dtype=float)
+    dwell_time_arr = t_irr_arr.copy()
+    dwell_time = symbols('dwell_time')
+    pulse_length_list = []
+    
+    for num_ind, num in enumerate(num_pulses):
+        pulse_length = active_burn_time / num
+        pulse_length_list.append(pulse_length)
+        for duty_cycle_ind, duty_cycle in enumerate(duty_cycle_list):
+            dwell_time_eq = Eq(pulse_length / (pulse_length + dwell_time), duty_cycle)
+            dwell_time_sol = solve((dwell_time_eq),(dwell_time))
+            dwell_time_arr[num_ind, duty_cycle_ind] = dwell_time_sol[0]
+            # The total irradiation time can be calculated once the dwell time has been found
+            t_irr = pulse_length * num + dwell_time_sol[0] * (num - 1)
+            t_irr_arr[num_ind, duty_cycle_ind] = t_irr   
+    return pulse_length_list, dwell_time_arr, t_irr_arr
+             
+def write_out_adf(inputs):
+    #runs_list = [inputs['runs_100_4y'], inputs['runs_90_4y'], inputs['runs_50_4y'], inputs['runs_25_4y']]
+    #start with one dictionary of runs first
+    runs_list = [inputs['runs_100_4y']]
+    for runs_idx, runs in enumerate(runs_list):
+        lib = aop.DataLibrary()
+        adf = aop.DataLibrary.make_entries(lib, runs)
+    return adf
+
+def store_flux_lines(flux_lines, num_blocks):
+    energy_bins = openmc.mgxs.GROUP_STRUCTURES['VITAMIN-J-175'] 
+    bin_widths = []
+    for bin_index in range(len(energy_bins) - 1):
+        bin_width = energy_bins[bin_index + 1] - energy_bins[bin_index]
+        bin_widths.append(bin_width)
+
+    all_entries = []
+    for flux_line in flux_lines:
+        if flux_line.strip(): #if the current line is not blank
+            all_entries.extend(flux_line.split())
+    all_entries = np.array(all_entries, dtype=float)
+    flux_array = all_entries.reshape(num_blocks, len(bin_widths))
+    return bin_widths, flux_array
+
+def normalize_flux_spectrum(flux_array, bin_widths, num_blocks): #flux spectrum shape
+    norm_flux_array = flux_array.copy()
+    total_flux = np.sum(flux_array, axis=1) #sum over the bin widths of flux array
+
+    for zone_idx in range(num_blocks):
+        norm_flux_array[zone_idx,:] = (norm_flux_array[zone_idx,:] / bin_widths) * (1 / total_flux[zone_idx])  
```

Comment:
Is there a numpy way to do this without a loop?  Something like this, but maybe needs some checking/testing
```suggestion
    norm_flux_array = (norm_flux_array.T / total_flux).T / bin_widths
```